### ISSUE TITLE
`stdin.read` (and `stdin.readSync`) corrupt non-ASCII input on Windows

### ISSUE THREAD

**Issue by lionel-rowe** [denoland/deno#18240]
`stdin.read` (and `stdin.readSync`) corrupt non-ASCII input on Windows.

To reproduce:

```js
const c = new Uint8Array(6); const n = await Deno.stdin.readSync(c); console.log(c)
```

Then, enter a non-ASCII character. The resulting bytes will be corrupted on Windows.

Examples,with trailing LF/CRLF/null bytes truncated:

```
ÿ
# expected [195, 191]; got [152] (U+0098 START OF STRING)
Ā
# expected [196, 128]; got [65] (A)
ā
# expected [196, 129]; got [97] (a)
啊
# expected [229, 149, 138]; got [63] (?)
🦄
# expected [240, 159, 166, 132]; got [63, 63] (??)
```

Results on Linux are as expected.

**Comment by njhanley** [comment]
This stems from Deno [reading directly from console input as a file][1], which uses the console's [current code page][2] rather than Unicode (see [High-Level Console I/O][3]). In the example, 'ÿ' maps to 152 in code page 437 (OEM-US).

Instead [`ReadConsoleW`][4] should be used, as in Rust's [`std::io::Stdin`][5].

@dsherret Is there a reason Deno doesn't use Rust/Tokio's stdio implementation? If not, would a PR be welcome?

[1]: https://github.com/denoland/deno/blob/3cd7abf73fa104526508984daef54bbb8e120310/ext/io/lib.rs#L63-L67
[2]: https://learn.microsoft.com/en-us/windows/console/console-code-pages
[3]: https://learn.microsoft.com/en-us/windows/console/high-level-console-i-o
[4]: https://learn.microsoft.com/en-us/windows/console/readconsole
[5]: https://github.com/rust-lang/rust/blob/3a8a131e9509c478ece1c58fe0ea2d49463d2300/library/std/src/sys/windows/stdio.rs

**Comment by dsherret** [comment]
Yes, a PR would be welcome. I believe it should use `std::io::stdin` here when `StdFileResourceKind::Stdin`:

https://github.com/denoland/deno/blob/3cd7abf73fa104526508984daef54bbb8e120310/ext/io/lib.rs#L398-L400

Similar to how it does this for write:

https://github.com/denoland/deno/blob/3cd7abf73fa104526508984daef54bbb8e120310/ext/io/lib.rs#L377-L383

**Comment by Mqxx** [comment]
Hey, I just stumbled across this issue having the same problem. Characters like `ÄÖÜ` (non ASCII) are corrupted. Any update on when this gets fixed?

Thanks
